x-pack/metricbeat/module/awsfargate/task_stats/task_stats.go

```diff
@@ -0,0 +1,203 @@
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package task_stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/docker/docker/api/types"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/awsfargate"
+)
+
+var metricsetName = "task_stats"
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(awsfargate.ModuleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	*awsfargate.MetricSet
+	logger *logp.Logger
+}
+
+// Stats is a struct represents information regarding a container
+type Stats struct {
+	Time         common.Time
+	Container    *container
+	cpuStats     cpuStats
+	memoryStats  memoryStats
+	networkStats []networkStats
+	blkioStats   blkioStats
+}
+
+// TaskMetadata is an struct represents response body from ${ECS_CONTAINER_METADATA_URI_V4}/task
+type TaskMetadata struct {
+	Cluster    string       `json:"Cluster"`
+	TaskARN    string       `json:"TaskARN"`
+	Family     string       `json:"Family"`
+	Revision   string       `json:"Revision"`
+	Containers []*container `json:"Containers"`
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	logger := logp.NewLogger(metricsetName)
+	metricSet, err := awsfargate.NewMetricSet(base)
+	if err != nil {
+		return nil, fmt.Errorf("error creating %s metricset: %w", metricsetName, err)
+	}
+
+	return &MetricSet{
+		MetricSet: metricSet,
+		logger:    logger,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	ecsURI, ok := os.LookupEnv("ECS_CONTAINER_METADATA_URI_V4")
+	if !ok {
+		err := fmt.Errorf("lookup $ECS_CONTAINER_METADATA_URI_V4 failed")
+		m.logger.Error(err)
+		return err
+	}
+
+	taskStatsEndpoint := fmt.Sprintf("%s/task/stats", ecsURI)
```

Comment:
nit: I think we could define these two paths as consts in the very begginging of the file so as to have easier access to them. wdyt?